google/auth/aws.py

```diff
@@ -637,33 +661,54 @@ def __init__(
             *args,
             **kwargs
         )
-        credential_source = credential_source or {}
-        self._target_resource = audience
-        environment_id = credential_source.get("environment_id") or ""
-        self._aws_security_credentials_supplier = _DefaultAwsSecurityCredentialsSupplier(
-            credential_source
-        )
-        self._cred_verification_url = credential_source.get(
-            "regional_cred_verification_url"
-        )
+        if credential_source is None and aws_security_credentials_supplier is None:
+            raise exceptions.InvalidValue(
+                "A valid credential source or AWS security credentials supplier must be provided."
+            )
+        if (
+            credential_source is not None
+            and aws_security_credentials_supplier is not None
+        ):
+            raise exceptions.InvalidValue(
+                "AWS credential cannot have both a credential source and an AWS security credentials supplier."
+            )
 
-        # Get the environment ID. Currently, only one version supported (v1).
-        matches = re.match(r"^(aws)([\d]+)$", environment_id)
-        if matches:
-            env_id, env_version = matches.groups()
+        if aws_security_credentials_supplier:
+            self._aws_security_credentials_supplier = aws_security_credentials_supplier
+            self._cred_verification_url = (
+                _DEFAULT_AWS_REGIONAL_CREDENTIAL_VERIFICATION_URL
+            )
+            _LOGGER.info(
```

Comment:
I think this qualifies as a debug level log.

Reply:
I added this based off your previous comment - https://github.com/googleapis/google-auth-library-python/pull/1484#discussion_r1508223845, I may have not understood what you are asking for

Reply:
Sorry my comment was vague - I meant a docstring block like [this](https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/credentials.py#L383) to visually elevate this info in the documentation.

Reply:
Ahh, makes sense, I will just remove the logging blocks then and add that instead - thanks